src/main/java/bio/terra/service/iam/IamService.java

```diff
@@ -186,6 +186,16 @@ public void deleteSnapshotResource(AuthenticatedUserRequest userReq, UUID snapsh
         }
     }
 
+    public Map<IamRole, String> retrievePolicyEmails(AuthenticatedUserRequest userReq,
+                                              IamResourceType iamResourceType,
+                                              UUID resourceId) {
+        try {
```

Comment:
It's not a lot of duplicated code but it may be worth refactoring anyway. Something like
```

    private interface Func<T> {
        T call() throws InterruptedException;
    }

    private <T> T callWithCatch(Func<T> func) {
        try {
            return func.call();
        } catch (InterruptedException ex) {
            throw new IamUnavailableException("service unavailable");
        }
    }

    public Map<IamRole, String> retrievePolicyEmails(AuthenticatedUserRequest userReq,
                                                     IamResourceType iamResourceType,
                                                     UUID resourceId) {
        return callWithCatch(() -> iamProvider.retrievePolicyEmails(userReq, iamResourceType, resourceId));
    }
```

If we ever change the throw type it would make things a lot easier. Methods that return void are a little awkward through and would need something like:

```
    public void deleteSnapshotResource(AuthenticatedUserRequest userReq, UUID snapshotId) {
        callWithCatch(() -> {
            iamProvider.deleteSnapshotResource(userReq, snapshotId);
            return 0;
        });
    }
```

Reply:
Ooh interesting!  I may do that separately but let me chew on that